### PR TITLE
Fixing issue with hanging UI on lots of IoT data

### LINKED ISSUE #444: device-explorer hangs on receiving high volume of events
when processing a large number of events from IOT hub, the device-explorer UI hangs. You have to kill it with task manager

### PR DESCRIPTION
**Pull request by bospoort**
# Checklist
- [X] I have read the [contribution guidelines](https://github.com/Azure/azure-iot-sdk-csharp/blob/master/.github/CONTRIBUTING.md).
- [x] I added or modified the existing tests to cover the change (we do not allow our test coverage to go down).
- [x] This pull-request is submitted against the `master` branch.

# Description of the changes
Moved receiving events from eventhub on a background thread in a Task. Issue was that the UI hangs when there are a lot of events coming in. 

# Reference/Link to the issue solved with this PR (if any)
Fixes #444

### CODE REVIEW

File: tools/DeviceExplorer/DeviceExplorer/MainForm.cs [hunk 1]
```diff
@@ -532,55 +535,9 @@ private void listButton_Click(object sender, EventArgs e)
...
-                {
-                    ct.ThrowIfCancellationRequested();
-
-                    var eventData = await eventHubReceiver.ReceiveAsync(TimeSpan.FromSeconds(1));
-
-                    if (eventData != null)
-                    {
-                        var data = Encoding.UTF8.GetString(eventData.GetBytes());
-                        var enqueuedTime = eventData.EnqueuedTimeUtc.ToLocalTime();
+                Func<bool> processor = new Func<bool>(()=>ProcessOutput(eventHubReceiver, selectedDevice, ct));
```
Comment (CIPop) [inline tools/DeviceExplorer/DeviceExplorer/MainForm.cs]: nit: spacing `(() => ProcessOutput`

**Comment by CIPop** [comment]
I've updated the description to auto-resolve #444 once this gets merged.

**Comment by bospoort** [comment]
I made all the requested changes. Thanks

**Comment by tameraw** [comment]
@bospoort thank you for your contribution to our open-sourced project!  Please help us improve by filling out this 2-minute [customer satisfaction survey](https://www.surveymonkey.com/r/Q5PLZ3V).

**Pull request merged**